foxpuppet/windows/browser/bookmarks/bookmark.py

```diff
@@ -0,0 +1,212 @@
+from selenium.common.exceptions import NoSuchElementException
+from selenium.webdriver.remote.webelement import WebElement
+from selenium.webdriver.common.keys import Keys
+from foxpuppet.windows.browser.navbar import NavBar
+from typing import TYPE_CHECKING, Optional, TypedDict, List
+
+
+class BookmarkData(TypedDict):
+    """Bookmark properties."""
+
+    name: str
+    url: str
+    tags: Optional[List[str]]
+    keyword: Optional[str]
+
+
+class Bookmark(NavBar):
+    """Handles Bookmark operations in Firefox."""
+
+    if TYPE_CHECKING:
+        from foxpuppet.windows.browser.window import BrowserWindow
+
+    @staticmethod
+    def create(window: "BrowserWindow", root: WebElement) -> "Bookmark":
+        """Create a bookmark object.
+
+        Args:
+            window (:py:class:`BrowserWindow`): Window object this bookmark appears in
+            root (:py:class:`~selenium.webdriver.remote.webelement.WebElement`): WebDriver element object for bookmark
+
+        Returns:
+            :py:class:`Bookmark`: Bookmark instance
+        """
+        with window.selenium.context(window.selenium.CONTEXT_CHROME):
+            return Bookmark(window, root)
+
+    @property
+    def is_bookmarked_star(self) -> bool:
+        """Checks if the current page is bookmarked using the star button.
+
+        Returns:
+            bool: True if the page is bookmarked, False otherwise.
+        """
+        with self.selenium.context(self.selenium.CONTEXT_CHROME):
+            star_button_image = self.selenium.find_element(
+                *BookmarkLocators.STAR_BUTTON_IMAGE
+            )
+            return star_button_image.get_attribute("starred") == "true"
+
+    @property
+    def is_bookmarked_menu(self) -> bool:
+        """Checks if the current page is bookmarked using the main bookmark menu.
+
+        Returns:
+            bool: True if the page is bookmarked, False otherwise.
+        """
+        current_page_title = self.selenium.title
+        with self.selenium.context(self.selenium.CONTEXT_CHROME):
+            toolbar_element = self.selenium.find_element(
+                *BookmarkLocators.NAVIGATOR_TOOLBOX
+            )
+            self.actions.context_click(toolbar_element).perform()
+
+            menu_bar = self.selenium.find_element(*BookmarkLocators.MENU_BAR)
+            menu_bar.click()
+
+            bookmark_menu = self.selenium.find_element(
+                *BookmarkLocators.MAIN_MENU_BOOKMARK
+            )
+            if bookmark_menu is not None:
+                bookmark_menu.click()
+                menu_items = bookmark_menu.find_elements(
+                    By.CSS_SELECTOR, "menuitem.bookmark-item"
+                )
+                for item in menu_items:
+                    item_label = item.get_attribute("label")
+                    if item_label and item_label.lower() in current_page_title.lower():
+                        return True
+        return False
+
+    def add(self) -> None:
```

Comment:
What's the difference with `add()` or `add_bookmark()`?

Reply:
I should probably use a flag instead and combine both into one method so it isn't confusing